app/api/coreApi.js

```diff
@@ -905,6 +913,82 @@ function getBlockByHashWithTransactions(blockHash, txLimit, txOffset) {
 	});
 }
 
+const getBlockListDefaultArgs = {
+	limit: config.site.browseBlocksPageSize,
+	offset: 0,
+	sort: 'desc'
+};
+
+function getBlockList(args)
+{
+	args = Object.assign(Object.assign({}, getBlockListDefaultArgs), args);
+
+	return new Promise(function(resolve, reject) {
+		getBlockchainInfo().then(function(getblockchaininfo) {
+			var sortDesc = args.sort == 'desc';
+
+			// Get all the block heights we will display
+			var blockHeights = Array.from({length: args.limit})
+				.map((_, i) => sortDesc ? (getblockchaininfo.blocks - args.offset - i) : (args.offset + i))
+				.filter(h => h >= 0 && h <= getblockchaininfo.blocks);
+
+			// hack: default regtest node returns getblockchaininfo.blocks=0, despite having a genesis block
+			if (global.activeBlockchain == "regtest" && blockHeights.length < 1)
+				blockHeights.push(0);
+
+			// Check if we can fetch an extra block height for time difference calculation
+			var hasExtraElement = false;
+			var extraElement = sortDesc ? getblockchaininfo.blocks - args.offset - args.limit : args.offset - 1;
+			if (extraElement >= 0) {
+				if (sortDesc)
+					blockHeights.push(extraElement);
+				else
+					blockHeights.unshift(extraElement);
+				hasExtraElement = true;
+			}
+			
+			Promise.all(blockHeights.map(h => rpcApi.getBlockHash(h))).then(function(blockHashes) {
```

Comment:
Do not cache `getblockhash` to prevent caching issues during re-orgs.

Reply:
good idea.